lib/galaxy/tools/parameters/validation.py

```diff
@@ -58,7 +58,7 @@ def __init__( self, message, expression ):
         self.expression = expression
 
     def validate( self, value, trans=None ):
-        if re.match( self.expression, value ) is None:
+        if re.match( self.expression, value or '' ) is None:
```

Comment:
Maybe checking [here]( https://github.com/galaxyproject/galaxy/blob/02187afb57a3b212c4e9a05b153e117bc8778866/lib/galaxy/tools/parameters/__init__.py#L99) that we are in a tool building context would be better?